primitives/src/smart_extension.rs

```diff
@@ -0,0 +1,28 @@
+use codec::{Decode, Encode};
+use sp_std::prelude::Vec;
+
+/// Smart Extension types
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum SmartExtensionTypes {
```

Comment:
Singular, `SmartExtensionType`, since an object of this type represents a single smart extension type rather than many.